src/Entity/ItemId.php

```diff
@@ -95,16 +96,18 @@ public function unserialize( $serialized ) {
 	 * serialization. Not doing so in new code requires special justification.
 	 *
 	 * @param int|float|string $numericId
+	 * @param string $repositoryName, defaults to an empty string (local repository)
 	 *
 	 * @return self
 	 * @throws InvalidArgumentException
 	 */
-	public static function newFromNumber( $numericId ) {
+	public static function newFromNumber( $numericId, $repositoryName = '' ) {
 		if ( !is_numeric( $numericId ) ) {
 			throw new InvalidArgumentException( '$numericId must be numeric' );
 		}
+		RepositoryNameAssert::assertParameterIsValidRepositoryName( $repositoryName, '$repositoryName' );
```

Comment:
@jakobw: as @brightbyte pointed out above, given you only have a number you want to create an entity id instance of, and the repository name (possibly "empty" one), it all simply limits you to only have ids like "foo:Q1234". `ItemId::newFromNumber` is kind of by its definition limited and not suppose to cover all that ItemId's constructor does.